pkg/skaffold/deploy/cloudrun/accessor.go

```diff
@@ -0,0 +1,175 @@
+}
+
+type forwardedResource struct {
+	name    RunResourceName
+	cancel  context.CancelFunc
+	started bool
+	port    int
+}
+
+type forwarder interface {
+	// Start initiates the forwarder's operation. It should not return until any ports have been allocated.
+	Start(ctx context.Context, out io.Writer) error
+	Stop()
+}
+
+// RunAccessor is an access.Accessor for Cloud Run resources
+// It uses `gcloud run proxy``to enable port forwarding for Cloud Run. This makes it easier to call IAM-protected Cloud Run services
+// by going through localhost. In order to set up forwarding, the services must have their ingress setting set to "all", gcloud  must be
+// installed and on the path, and the currently configured gcloud user has run.services.invoke permission on the services being proxied
+type RunAccessor struct {
+	resources  *resourceTracker
+	forwarders []forwarder
+	singleRun  singleflight.Group
+	label      string
+}
+
+// NewAccessor creates a new RunAccessor to port forward Cloud Run services
+func NewAccessor(cfg Config, label string) *RunAccessor {
+	var forwarders []forwarder
+	resources := &resourceTracker{forwardedPorts: &util.PortSet{}}
+	options := cfg.PortForwardOptions()
+	if options.ForwardServices(cfg.Mode()) {
+		forwarders = append(forwarders, &runProxyForwarder{resources: resources})
+	}
+	return &RunAccessor{
+		resources: resources, forwarders: forwarders, label: label, singleRun: singleflight.Group{}}
+}
+
+// AddResource tracks an additional resource to port forward
+func (r *RunAccessor) AddResource(resource RunResourceName) {
+	r.resources.resources = append(r.resources.resources, forwardedResource{name: resource, started: false})
+}
+
+// Start begins port forwarding for the tracked Cloud Run services.
+func (r *RunAccessor) Start(ctx context.Context, out io.Writer) error {
+	if r == nil {
+		return nil
+	}
+	_, err, _ := r.singleRun.Do(r.label, func() (interface{}, error) {
+		return struct{}{}, r.start(ctx, out)
+	})
+	return err
+}
+
+func (r *RunAccessor) start(ctx context.Context, out io.Writer) error {
+	for _, forwarder := range r.forwarders {
+		if err := forwarder.Start(ctx, out); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Stop terminates port forwarding for all tracked Cloud Run resources.
+func (r *RunAccessor) Stop() {
+	for _, forwarder := range r.forwarders {
+		forwarder.Stop()
+	}
+}
+
+func findGcloud() bool {
+	cmd := exec.Command("gcloud", "--version")
+	return cmd.Run() == nil
+}
+
+type runProxyForwarder struct {
+	resources *resourceTracker
+}
+
+func (r *runProxyForwarder) Start(ctx context.Context, out io.Writer) error {
+	if !gcloudInstalled() {
+		output.Red.Fprintln(out, "gcloud not found on path. Unable to set up Cloud Run port forwarding")
+		return sErrors.NewError(fmt.Errorf("gcloud not found"), &proto.ActionableErr{ErrCode: proto.StatusCode_PORT_FORWARD_RUN_GCLOUD_NOT_FOUND})
+	}
+	for _, resource := range r.resources.resources {
+		if resource.port == 0 {
```

Comment:
not sure the cause but resource.port always seems to be 0 even if the value is set via the `portForward` schema

Reply:
Added code to plumb through port overrides. It doesn't support changing the address, although I can see about adding that to gcloud as well. 

I did re-use the "service" resource type here. I can add a new type for cloud run services, but then I'd have to make sure the kubectl forwarder doesn't error out on cloud run resources.